### PR TITLE
chore(parser): fetch CSV instead of HTML in CA-AB

### PR DESCRIPTION
**Pull request by kruschk**
The AESO ETS website now exposes a CSV interface in addition to HTML;
modify the CA-AB parser to take advantage of the simpler API.

Refs: #3626

### CODE REVIEW

File: parsers/CA_AB.py [hunk 1]
```diff
@@ -1,161 +1,153 @@
...
-    return [data[k] for k in sorted(data.keys())]
-
-
-def fetch_exchange(zone_key1='CA-AB', zone_key2='CA-BC', session=None, target_datetime=None, logger=None) -> dict:
-    """Requests the last known power exchange (in MW) between two countries."""
+        raise NotImplementedError('Currently unable to scrape historical data')
+    session = session or requests.Session()
+    response = session.get(f'{URL_STRING}/SMPriceReportServlet',
+                           params={'contentType': 'csv'})
+    return [
```
Comment (kruschk) [inline parsers/CA_AB.py:67]: The API serves the hourly prices in reverse-chronological order; I'm not sure if they should be reversed, or whether the order matters at all.

File: parsers/CA_AB.py [hunk 2]
```diff
@@ -1,161 +1,153 @@
...
+        'source': URL.netloc,
     }
 
 
-def fetch_price(zone_key='CA-AB', session=None, target_datetime=None, logger=None) -> dict:
-    """Requests the last known power price of a given country."""
+def fetch_price(zone_key=DEFAULT_ZONE_KEY,
+                session=None,
+                target_datetime=None,
+                logger=None) -> dict:
```
Comment (kruschk) [inline parsers/CA_AB.py:60]: The return type should be `list<dict>`, not `dict`.

File: parsers/CA_AB.py [hunk 3]
```diff
@@ -1,161 +1,153 @@
...
+            'zoneKey': zone_key,
+        } for row in csv.reader(response.text
+                                        .split('\r\n\r\n')[2]
+                                        .splitlines()[1:]) if row[1] != '-']
+
+
+def fetch_production(zone_key=DEFAULT_ZONE_KEY,
+                     session=None,
+                     target_datetime=None,
+                     logger=None) -> dict:
```
Comment (kruschk) [inline parsers/CA_AB.py:83]: `validation.validate` expects a valid `logger` argument, but it defaults to `None` here. I should change this to `logger=logging.getLogger(__name__)`.

File: parsers/CA_AB.py (hunk 1 above)
Comment (madsnedergaard) [reply parsers/CA_AB.py:68]: I couldn't tell from looking at the backend either, did you check how other parsers does it? I guess the ENTSO-E or FR should be trustworthy 🤔

Comment (kruschk) [reply parsers/CA_AB.py:68]: ENTSO-E and FR appear to return values in chronological (i.e., most recent last) order; however, neither parser explicitly sorts the results; the order is just a consequence of how the API presents the data. I tried shuffling the list with `random.shuffle`, and `poetry run test_parser CA-AB price` was successful no matter what order was presented to it. This leads me to believe that the order doesn't matter.

Comment (madsnedergaard) [reply parsers/CA_AB.py:68]: It probably doesn't, but thanks for looking into it :)

**Review by madsnedergaard — APPROVED** [review]
LGTM! 🥳

**Pull request merged**